src/StreamSelectLoop.php

```diff
@@ -21,11 +22,27 @@ class StreamSelectLoop implements LoopInterface
     private $writeStreams = [];
     private $writeListeners = [];
     private $running;
+    private $pcntl = false;
+    private $signals;
 
     public function __construct()
     {
         $this->futureTickQueue = new FutureTickQueue();
         $this->timers = new Timers();
+        $this->pcntl = extension_loaded('pcntl');
+        $this->signals = new SignalsHandler();
+        $this->signals->on('on', function ($signal) {
+            \pcntl_signal($signal, function ($signal) {
+                $this->signals->call($signal);
+            });
```

Comment:
That traps signal and installs a process-wide handler. I guess it should at least be cleaned up in something like `__destruct()`. Behavior with multiple loops needs to be defined somehow.